src/display/annotation_layer.js

```diff
@@ -760,7 +849,9 @@ var AnnotationLayer = (function AnnotationLayerClosure() {
           linkService: parameters.linkService
         };
         var element = annotationElementFactory.create(properties);
-        parameters.div.appendChild(element.render());
+        if (element.hasHtml()) {
```

Comment:
To me, it seems slightly backwards to first create the element, and *then* check if it's actually needed. With the current code, we don't do anything in the `hasHtml === false` case, which seems conceptually more correct.